### PR TITLE
[webui] Create comments via ajax requests

### PR DESCRIPTION
**Pull request by mschnitzer**
Comments will now be created via Ajax requests. The comment section is
reloaded every time, when a new comment has been added by the user.

Affects:
* Requests
* Packages
* Projects

### CODE REVIEW

**Comment by mschnitzer** [comment]
@ChrisBr please have a look at this as you were working on the comment deletion part.

**Comment by evanrolfe** [comment]
Is there a particular reason for making the comments ajax?

**Comment by mschnitzer** [comment]
@evanrolfe 

* We already using Ajax for comment deletion (as it was requested by coolo)
* You don't have to load the whole page again. Especially for requests this is important because sometimes people create requests with many target actions. Loading the whole page again would end up into executing many database queries again for nothing.

File: src/api/app/controllers/webui/comments_controller.rb [hunk 1]
```diff
@@ -5,11 +5,13 @@ class Webui::CommentsController < Webui::WebuiController
   def create
     comment = @commented.comments.new(permitted_params)
     User.current.comments << comment
+    @comment = Comment.new
```
Comment (evanrolfe) [inline src/api/app/controllers/webui/comments_controller.rb]: Why do we have two variables: `comment` and `@comment`? Whats the difference between the two?

Comment (mschnitzer) [reply src/api/app/controllers/webui/comments_controller.rb]: `@comment = Comment.new` is required inside the partial for the form construction. See: https://github.com/openSUSE/open-build-service/blob/master/src/api/app/views/webui/comment/_new.html.haml#L2

Comment (evanrolfe) [reply src/api/app/controllers/webui/comments_controller.rb]: Hmm, I think this is why we shouldn't use instance variables in partials... Ok can you please do either:
1. Use a local variable in the _new partial
or
2. Add a comment explaining why we set @comment here

**Comment by mschnitzer** [comment]
@evanrolfe I added a comment as you proposed.

**Comment by mschnitzer** [comment]
@ChrisBr ping

**Comment by ChrisBr** [comment]
If @evanrolfe approved the PR you don't need my review as well ;)

**Pull request merged**